src/watch/watcher.ts

```diff
@@ -1,3 +1,7 @@
+import ora from "ora";
 import AtomsServer from "../server";
 
-export type Watcher = (server: AtomsServer) => (path: string) => Promise<void>;
+export type Watcher = (
+  server: AtomsServer,
+  spinner: ora.Ora
```

Comment:
should the spinner be part of AtomsServer? as it is coupled with a server instance and is better not to pollute the watcher interface in case we want to support watchers from external repos

Reply:
I had to add it her so we can reset the spinner while building changed files and add the watch once again. This was the only way I could do it, do you have any other suggestions?